plugins/PostProcessingPlugin/scripts/AddCoolingProfile.py

```diff
@@ -0,0 +1,803 @@
+                    "type": "int",
+                    "default_value": 35,
+                    "minimum_value": 0,
+                    "maximum_value": 100,
+                    "unit": "%    ",
+                    "enabled": "fan_layer_or_feature == 'by_feature'"
+                },
+                "fan_bridge":
+                {
+                    "label": "Bridge %",
+                    "description": "Enter the fan percentage for any Bridging (whether it's used on not).",
+                    "type": "int",
+                    "default_value": 100,
+                    "minimum_value": 0,
+                    "maximum_value": 100,
+                    "unit": "%    ",
+                    "enabled": "fan_layer_or_feature == 'by_feature'"
+                },
+                "fan_combing":
+                {
+                    "label": "Fan 'OFF' during Combing:",
+                    "description": "When checked will set the fan to 0% for combing moves over 5 lines long in the gcode. When un-checked the fan speed during combing is whatever the previous speed is set to.",
+                    "type": "bool",
+                    "enabled": "fan_layer_or_feature == 'by_feature'",
+                    "default_value": true
+                },
+                "fan_feature_final":
+                {
+                    "label": "Final %",
+                    "description": "If you choose an 'End Layer' then this is the fan speed that will carry through to the end of the gcode file.  It will go into effect at the 'END' of your end layer.",
+                    "type": "int",
+                    "default_value": 35,
+                    "minimum_value": 0,
+                    "maximum_value": 100,
+                    "unit": "%    ",
+                    "enabled": "(int(fan_end_layer) != -1) and (fan_layer_or_feature == 'by_feature')"
+                },
+                "fan_enable_raft":
+                {
+                    "label": "Enable Raft Cooling",
+                    "description": "Enable the fan for the raft layers.  When enabled the Raft Fan Speed will continue until another Layer or Feature setting over-rides it.",
+                    "type": "bool",
+                    "default_value": false,
+                    "enabled": true
+                },
+                "fan_raft_percent":
+                {
+                    "label": "Raft Fan %:",
+                    "description": "Enter the percentage for the Raft.",
+                    "type": "int",
+                    "default_value": 35,
+                    "minimum_value": 0,
+                    "maximum_value": 100,
+                    "unit": "%    ",
+                    "enabled": "fan_enable_raft"
+                }
+            }
+        }"""
+
+    def execute(self, data):
+#Initialize variables that are buried in if statements.
+        T0_fan = " P0"; T1_fan = " P0"; T2_fan = " P0"; T3_fan = " P0"; is_multi_extr_print = True
```

Comment:
The support for additional fan circuits is complex but once you get above 1 then it really isn't any more difficult to add more.  I see this post processor being of greater value to industrial users doing large prints.  We both know that all the complaints will come for the dnd and "mini" crowd.
I waffled on an array or list.  In the end it was only a couple of lines of code difference.  It's already as long as War and Peace.and cutting a few lines out just didn't have much benefit.  It runs faster than I thought it would.

Reply:
I don't really mind much that it's limited. Just thinking about ways to communicate that to the user.

Sure, it's long now, and replacing _one_ manual iteration with a for-loop won't indeed matter much. Doing so consistently however, will make it a lot shorter.